src/tslint.ts

```diff
@@ -99,13 +98,19 @@ class Linter {
         return rules.some((r) => r.equals(rule));
     }
 
-    private computeFullOptions() {
-        let {configuration, rulesDirectory} = this.options;
-        if (configuration == null) {
-            configuration = DEFAULT_CONFIG;
+    private computeFullOptions(options: ILinterOptionsRaw): ILinterOptions {
+        if (!options || typeof options !== "object") {
```

Comment:
I actually liked what you had before where you threw an error if `typeof options !== "object"`. It's clearly a mistake if that happens, so let's throw an error! As far as if `options` is just null, then we can continue using all the defaults below